solidity/contracts/bank/Vault.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.4;
+
+import "../bank/Bank.sol";
+
+/// @title Bitcoin Bank Vault
+/// @notice Vault is a contract extended by applications built on top of the
+///         Bank. Vault is locking balances under its own address by
+///         transferring the balances to the Vault address and internally
+///         bookkeeping the amount of a locked balance per account. This way,
+///         each application has a separate Bitcoin balance and problems in
+///         a single application are not propagated to other applications or
+///         Bank depositors.
+contract Vault {
+    Bank public bank;
+
+    /// @notice The balance of a given account locked in the Vault.
+    ///         Zero by default.
+    mapping(address => uint256) public lockedBalance;
+
+    constructor(Bank _bank) {
+        require(
+            address(_bank) != address(0),
+            "Bank can not be the zero address"
+        );
+        bank = _bank;
+    }
+
+    /// @notice Transfers the given `amount` of the Bank balance from `spender`
+    ///         to the Vault and locks this balance under `spender`'s account.
+    /// @dev Vault must have an allowance for `spender`'s balance for at least
+    ///      `amount`.
+    function lockBalance(address spender, uint256 amount) internal {
+        require(
+            bank.balanceOf(spender) > amount,
```

Comment:
Shouldn't it be:
```
bank.balanceOf(spender) >= amount,
```
?

Reply:
Ooops 🙈 
2976a842552795dec2f56f34d621eb65d873ebf0